### ISSUE TITLE
descriptor.pb.cc: misleading-indentation

### ISSUE THREAD

**Issue by ignatenkobrain** [google/protobuf#1797]
``` cpp
google/protobuf/descriptor.pb.cc: In member function ‘virtual bool google::protobuf::FileOptions::IsInitialized() const’:
google/protobuf/descriptor.pb.cc:8762:3: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
   if (!_extensions_.IsInitialized()) return false;  return true;
   ^~
google/protobuf/descriptor.pb.cc:8762:53: note: ...this statement, but the latter is misleadingly indented as if it is guarded by the ‘if’
   if (!_extensions_.IsInitialized()) return false;  return true;
                                                     ^~~~~~
google/protobuf/descriptor.pb.cc: In member function ‘virtual bool google::protobuf::MessageOptions::IsInitialized() const’:
google/protobuf/descriptor.pb.cc:9722:3: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
   if (!_extensions_.IsInitialized()) return false;  return true;
   ^~
google/protobuf/descriptor.pb.cc:9722:53: note: ...this statement, but the latter is misleadingly indented as if it is guarded by the ‘if’
   if (!_extensions_.IsInitialized()) return false;  return true;
                                                     ^~~~~~
google/protobuf/descriptor.pb.cc: In member function ‘virtual bool google::protobuf::FieldOptions::IsInitialized() const’:
google/protobuf/descriptor.pb.cc:10416:3: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
   if (!_extensions_.IsInitialized()) return false;  return true;
   ^~
google/protobuf/descriptor.pb.cc:10416:53: note: ...this statement, but the latter is misleadingly indented as if it is guarded by the ‘if’
   if (!_extensions_.IsInitialized()) return false;  return true;
                                                     ^~~~~~
google/protobuf/descriptor.pb.cc: In member function ‘virtual bool google::protobuf::EnumOptions::IsInitialized() const’:
google/protobuf/descriptor.pb.cc:10957:3: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
   if (!_extensions_.IsInitialized()) return false;  return true;
   ^~
google/protobuf/descriptor.pb.cc:10957:53: note: ...this statement, but the latter is misleadingly indented as if it is guarded by the ‘if’
   if (!_extensions_.IsInitialized()) return false;  return true;
                                                     ^~~~~~
google/protobuf/descriptor.pb.cc: In member function ‘virtual bool google::protobuf::EnumValueOptions::IsInitialized() const’:
google/protobuf/descriptor.pb.cc:11339:3: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
   if (!_extensions_.IsInitialized()) return false;  return true;
   ^~
google/protobuf/descriptor.pb.cc:11339:53: note: ...this statement, but the latter is misleadingly indented as if it is guarded by the ‘if’
   if (!_extensions_.IsInitialized()) return false;  return true;
                                                     ^~~~~~
google/protobuf/descriptor.pb.cc: In member function ‘virtual bool google::protobuf::ServiceOptions::IsInitialized() const’:
google/protobuf/descriptor.pb.cc:11696:3: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
   if (!_extensions_.IsInitialized()) return false;  return true;
   ^~
google/protobuf/descriptor.pb.cc:11696:53: note: ...this statement, but the latter is misleadingly indented as if it is guarded by the ‘if’
   if (!_extensions_.IsInitialized()) return false;  return true;
                                                     ^~~~~~
google/protobuf/descriptor.pb.cc: In member function ‘virtual bool google::protobuf::MethodOptions::IsInitialized() const’:
google/protobuf/descriptor.pb.cc:12053:3: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
   if (!_extensions_.IsInitialized()) return false;  return true;
   ^~
google/protobuf/descriptor.pb.cc:12053:53: note: ...this statement, but the latter is misleadingly indented as if it is guarded by the ‘if’
   if (!_extensions_.IsInitialized()) return false;  return true;
                                                     ^~~~~~
```

**Comment by onlyjob** [comment]
`unittest.pb.cc` is affected as well.

This is a serious problem as it causes FTBFS on all architectures in Debian. Here is [build log](https://buildd.debian.org/status/fetch.php?pkg=protobuf&arch=i386&ver=3.0.0-1&stamp=1471906844).

**Comment by opoplawski** [comment]
Seen on Fedora Rawhide too:
```
google/protobuf/unittest_lite.pb.cc: In member function 'virtual bool protobuf_unittest::TestAllExtensionsLite::IsInitialized() const':
google/protobuf/unittest_lite.pb.cc:8544:3: error: this 'if' clause does not guard... [-Werror=misleading-indentation]
   if (!_extensions_.IsInitialized()) return false;  return true;
   ^~
google/protobuf/unittest_lite.pb.cc:8544:53: note: ...this statement, but the latter is misleadingly indented as if it is guarded by the 'if'
```

**Comment by fenwanderer** [comment]
Note that the `return false;  return true;` lines in `descriptor.pb.cc` seem to be independent of the fix proposed in #1698.  The `descriptor.pb.cc` file is already in the source and is not generated by `cpp_message.cc` at build time.  Patching `cpp_message.cc` should fix the FTBFS, but `descriptor.pb.cc` will still continue to give (non-fatal) warnings.  It will need to be fixed independently.

**Comment by fenwanderer** [comment]
This may have been fixed by 98835fb in v3.1.0.
